### PR TITLE
feat: timebook streamdeck plugin

### PR DESCRIPTION
**Pull request by inpaceee**
closes: #891

### CODE REVIEW

**Comment by inpaceee** [comment]
image

File: apps/stream-deck/net.progwise.timebook.sdPlugin/propertyInspector/tracking_pi.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  <head lang="en">
+    <meta charset="utf-8" />
+    <title>net.progwise.timebook PI</title>
+    <script src="https://cdn.jsdelivr.net/gh/geekyeggo/sdpi-components@v2/dist/sdpi-components.js"></script>
+  </head>
+  <body>
+    <sdpi-wrapper>
+      <sdpi-item label="Task">
+        <sdpi-select
+          setting="taskID"
```
Comment (PascalHelbig) [inline apps/stream-deck/net.progwise.timebook.sdPlugin/propertyInspector/tracking_pi.html:12]:
```suggestion
          setting="taskId"
```

and in TypeScript as well!

File: apps/stream-deck/net.progwise.timebook.sdPlugin/propertyInspector/tracking_pi.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          datasource="getProjects"
+          loading="Fetching projects..."
```
Comment (PascalHelbig) [inline apps/stream-deck/net.progwise.timebook.sdPlugin/propertyInspector/tracking_pi.html:14]:
```suggestion
          loading="Fetching tasks..."
```

File: apps/stream-deck/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "@tsconfig/node20": "^20.1.2",
+    "@types/node": "20.8.10",
+    "rollup": "^4.0.2",
+    "tslib": "^2.6.2",
+    "typescript": "^5.2.2"
+  },
+  "dependencies": {
+    "@elgato/streamdeck": "^0.3.0",
+    "date-fns": "^3.6.0",
+    "dotenv": "^16.0.3",
```
Comment (PascalHelbig) [inline apps/stream-deck/package.json:24]:
```suggestion
```

File: apps/stream-deck/src/actions/tracking.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  async onKeyDown(event: KeyDownEvent<TrackingSettings>): Promise<void> {
+    const { taskID } = await event.action.getSettings()
+    const client = await getClient()
+
+    try {
+      const response = await client.request<{
+        currentTracking?: { start: string; task: { id: string } }
+      }>(CurrentTrackingQueryDocument)
+
+      await (response.currentTracking && taskID === response.currentTracking?.task.id
```
Comment (PascalHelbig) [inline apps/stream-deck/src/actions/tracking.ts:155]:
```suggestion
      await (response.currentTracking && taskID === response.currentTracking.task.id
```

**Review by PascalHelbig — APPROVED** [review]
excited-button-gif-by-cartoon-hangover

**Review by YohannesTesfay — APPROVED** [review]

**Pull request merged**